vulnerable/helper.py

```diff
@@ -79,6 +79,31 @@ def remove_pathname(pathname) :
 
 # For nbgrader APIs
 
+def path_check(pathname) :
```

Comment:
Might want to make the signature `def path_check(pathname, basedir)` instead, since you aren't always submitting to cwd, but instead something like `NGSHARE_DIRECTORY/files/course/assignment/submission_abc/` (just as an idea). We can make sure that `basedir` can be trusted by using the numerical ID instead of the course / assignment name (`_id` in the database rather than `id`).